cmd/pinniped-proxy/Dockerfile

```diff
@@ -1,6 +1,6 @@
 # syntax = docker/dockerfile:1
 
-FROM rust:1.51 as builder
+FROM rust:1.53.0 as builder
```

Comment:
I forgot to mention you also have to change it here: https://github.com/kubeapps/kubeapps/blob/master/.circleci/config.yml#L11

(I hope to land my PR soon - including that change- soon)